### PR TITLE
chore: fix generated links

### PR DESCRIPTION
**Pull request by mkurapov**
### Context

After trying the [starlight-links-validator](https://starlight-links-validator.vercel.app/) in the Rafiki docs, it ended up outputting a lot of broken links as a result of the graphql generation:

Screenshot 2024-05-15 at 13 12 16

This was primarily a result of two issues:
- the plugin generates unique slugs for each link. However, we actually want to refer to the same heading object across pages. (e.g. we don't want to prepend a unique `-1` for linking to the same #grant object/anchor)
- Using camelCase `inputObjects` led to invalid links that should be pointed to `.../inputobjects` instead

### CODE REVIEW

File: src/converters/index.ts [hunk 1]
```diff
@@ -183,7 +183,7 @@ export const unions = {
 };
 
 export const inputObjects = {
-  id: "inputObjects",
+  id: "inputobjects",
```
Comment (mkurapov) [inline src/converters/index.ts:186]: we could also just lowercase the converter.id when we call `getRelativeTypeUrl` -> 

```ts
return `${converter.id}#${sluggify(baseType.name)}`
```

but this is straightforward as well?

File: src/converters/parseMarkdown.ts [hunk 2]
```diff
@@ -1,20 +1,20 @@
-import { marked } from "marked";
```
Comment (mkurapov) [inline src/converters/parseMarkdown.ts:1]: file was causing TS issues for me

`// marked's types are buggy` indeed

File: src/getRelativeTypeUrl.ts [hunk 3]
```diff
@@ -4,11 +4,10 @@ import {
   isListType,
   isNonNullType,
 } from "graphql";
-import Slugger from "github-slugger";
+import { slug } from "github-slugger";
```
Comment (mkurapov) [inline src/getRelativeTypeUrl.ts:7]: `slug` fn does _not_ generate unique ids

File: pnpm-lock.yaml [hunk 4]
```diff
@@ -0,0 +1,4047 @@
+lockfileVersion: '6.0'
```
Comment (mkurapov) [inline pnpm-lock.yaml]: @huijing was there a reason for not committing this originally?

Comment (huijing) [reply pnpm-lock.yaml]: this forces people to use pnpm but if they wanted to use some other package manager they would run into problems. are we mandating that everyone use pnpm for everything?

Comment (mkurapov) [reply pnpm-lock.yaml]: > this forces people to use pnpm

Only for the developers of this plugin. We can remove the fill & add it to .gitignore if you prefer using something else

**Review by huijing — APPROVED** [review]

**Comment by huijing** [comment]
Tested with local copy of Rafiki and https://github.com/HiDeoo/starlight-links-validator. Generated links are no longer the issue and the plugin is detecting actual mistypes in the documentation.
Bumped version to 0.3.0, please update Rafiki accordingly.

**Pull request merged**